vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/select_autocomplete.tsx

```diff
@@ -25,28 +27,26 @@ export default function SelectAutocompleteInput(
   const [inputState, setInputState] = React.useState(dispValue(value));
   const [options, setOptionsFull] = useState({
     filtered: options_in,
-    display: options_in
+    display: [...options_in].splice(0,maxOptions)
     });
   function setOptions(filteredOptions: string[]) {
     setOptionsFull({
       filtered: filteredOptions,
       display: [...filteredOptions].splice(0,maxOptions)})
   }
-  const getOptionsDelayed = useCallback(
-    debounce((text, options_in, callback) => {
-      getOptionsAsync(text, [...options_in]).then(callback);
-    }, 200),
-    [],
-  );
+  const [getOptionsDelayed] = useAsyncCallback(function*(text: string, options_in: string[], callback: Function) {
+    const options = yield getOptionsAsync(text, [...options_in]);
+    callback(options)
+  }, [])
   
   useEffect(() => {
     // Shown from all options when user has made their selection (current text = current value)
     const query = (inputState!=value) ? inputState : ''
     
     if (options_in.length>1000) {
-      setLoadingOptions(true); setOptions([]);
```

Comment:
This `setOptions([])` was a hangover from the example I'd used to set up our self-filtering of options.  Not needed anymore due to management of `loadingOptions` state variable given to `loading` prop below.